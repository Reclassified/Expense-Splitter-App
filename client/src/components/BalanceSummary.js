import React from 'react';

const BalanceSummary = ({ summary }) => (
  <div className="summary-section">
    <h3>Group Summary</h3>
    <div className="summary-stats">
      <div className="stat-item">
        <span className="stat-label">Total Expenses:</span>
        <span className="stat-value">${summary.totalExpenses.toFixed(2)}</span>
      </div>
      <div className="stat-item">
        <span className="stat-label">Total Paid:</span>
        <span className="stat-value">${summary.totalPaid.toFixed(2)}</span>
      </div>
      <div className="stat-item">
        <span className="stat-label">Status:</span>
        <span
          className={`stat-value ${summary.isBalanced ? 'balanced' : 'unbalanced'}`}
        >
          {summary.isBalanced ? 'Balanced' : 'Unbalanced'}
        </span>
      </div>
    </div>
  </div>
);

export default BalanceSummary;
