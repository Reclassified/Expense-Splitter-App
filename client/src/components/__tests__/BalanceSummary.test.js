import React from 'react';
import { render } from '@testing-library/react';
import BalanceSummary from '../BalanceSummary';

describe('BalanceSummary', () => {
  const summary = {
    totalExpenses: 123.45,
    totalPaid: 100.0,
    isBalanced: false,
  };

  it('renders correctly and matches snapshot', () => {
    const { container } = render(<BalanceSummary summary={summary} />);
    expect(container).toMatchSnapshot();
  });

  it('shows correct values and status', () => {
    const { getByText } = render(<BalanceSummary summary={summary} />);
    expect(getByText('Total Expenses:')).toBeInTheDocument();
    expect(getByText('$123.45')).toBeInTheDocument();
    expect(getByText('Total Paid:')).toBeInTheDocument();
    expect(getByText('$100.00')).toBeInTheDocument();
    expect(getByText('Status:')).toBeInTheDocument();
    expect(getByText('Unbalanced')).toBeInTheDocument();
  });
}); 