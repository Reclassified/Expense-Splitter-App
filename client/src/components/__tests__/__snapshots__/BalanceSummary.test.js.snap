// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceSummary renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="summary-section"
  >
    <h3>
      Group Summary
    </h3>
    <div
      class="summary-stats"
    >
      <div
        class="stat-item"
      >
        <span
          class="stat-label"
        >
          Total Expenses:
        </span>
        <span
          class="stat-value"
        >
          $
          123.45
        </span>
      </div>
      <div
        class="stat-item"
      >
        <span
          class="stat-label"
        >
          Total Paid:
        </span>
        <span
          class="stat-value"
        >
          $
          100.00
        </span>
      </div>
      <div
        class="stat-item"
      >
        <span
          class="stat-label"
        >
          Status:
        </span>
        <span
          class="stat-value unbalanced"
        >
          Unbalanced
        </span>
      </div>
    </div>
  </div>
</div>
`;
